src/parseUrl/parseUrl.js

```diff
@@ -0,0 +1,133 @@
+  if (url.length === 1) {
+    return {
+      isError: true,
+      response: {
+        errors: [
+          {
+            message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+          },
+          {
+            message:
+              '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+          },
+          {
+            message:
+              "This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.",
+          },
+        ],
+      },
+    };
+  }
+
+  let result = {};
+
+  if (!url.includes("?")) {
+    result = {
+      isError: true,
+      response: {
+        errors: [
+          {
+            message:
+              '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+          },
+        ],
+      },
+    };
+
+    return result;
+  }
+
+  const parts = url.split("?");
+
+  if (parts[0].length < 2) {
+    result = {
+      isError: true,
+      response: {
+        errors: result?.response?.errors
+          ? [
+              ...result.response.errors,
+              {
+                message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+              },
+            ]
+          : [
+              {
+                message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+              },
+            ],
+      },
+    };
+  }
+
+  const validCases = ["SNAKE", "KEBAB", "CAMEL", "PASCAL", "UPPER"];
+  const originalText = parts[0].slice(1);
+  const params = new URLSearchParams(parts[1]);
+  const targetCase = params.get("toCase");
+
+  console.log(targetCase);
+
+  if (!targetCase) {
+    result = {
+      isError: true,
+      response: {
+        errors: result?.response?.errors
+          ? [
+              ...result.response.errors,
+              {
+                message:
+                  '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+              },
+            ]
+          : [
+              {
+                message:
+                  '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+              },
+            ],
+      },
+    };
+  } else if (!validCases.includes(targetCase.toUpperCase())) {
+    result = {
+      isError: true,
+      response: {
+        errors: result?.response?.errors
+          ? [
+              ...result.response.errors,
+              {
+                message:
+                  "This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.",
+              },
+            ]
+          : [
+              {
+                message:
+                  "This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.",
+              },
+            ],
+      },
+    };
+  }
```

Comment:
Looks too complicated. We can create a separate object with error messages and create an error array. Then, we can check all possible options and push an error message to our array. In the end, we can check the array length. If it's not empty, we can send a status code of 400 and finish our function.

example: 

```
    if (!originalText) {
      errors.push({ message: errorMessages.noOriginalText });
    }
```

```
    if (errors.length) {
      res.statusCode = 400;
      res.statusMessage = 'Bad request';

      res.end(JSON.stringify({ errors }));

      return;
    }
```

Reply:
Done